### PR TITLE
feature/512: Use password history check

### PR DESCRIPTION
**Pull request by zamuzakki**
This is PR for #512 
Screenrecord when changing password:
[Kazam_screencast_00094.webm](https://github.com/kartoza/miniSASS/assets/7352963/ed74aa0d-d654-4a4f-8560-7b3d07154f09)

### CODE REVIEW

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 6 min 40 sec

File: django_project/minisass_authentication/serializers.py [hunk 1]
```diff
@@ -138,6 +138,12 @@ def validate_password(self, value):
         if all(remaining_requirements.values()):
             missing_criteria = ', '.join([key for key, val in remaining_requirements.items() if val])
             raise serializers.ValidationError(f'Missing password criteria: {missing_criteria}')
+
+        is_password_used = PasswordHistory.is_password_used(
+            self.context['user'], value
+        )
+        if is_password_used:
+            raise serializers.ValidationError('Password has been used')
```
Comment (tinashechiraya) [inline django_project/minisass_authentication/serializers.py:146]: This password has been used before. Please choose a new and unique password.

Oops! The password you entered has been used before. Kindly select a different password for better security.

would these be more informative ?

**Review by tinashechiraya — APPROVED** [review]
@zamuzakki maybe you can make that change later but I am merging now

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 6 min 2 sec

File: django_project/minisass_authentication/serializers.py (hunk 1 above)
Comment (zamuzakki) [reply django_project/minisass_authentication/serializers.py:146]: I'll update to this

> This password has been used before. Please choose a new and unique password.

Comment (zamuzakki) [reply django_project/minisass_authentication/serializers.py:146]: Updated in: https://github.com/kartoza/miniSASS/pull/570

**Pull request merged**